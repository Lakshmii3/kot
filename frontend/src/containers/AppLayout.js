import React, { Component } from 'react';
import {
    BrowserRouter as Router,
    Route,
    Link,
    Switch
} from 'react-router-dom';

import { Layout, Menu, Icon } from 'antd';

import GameboardLayout from './Gameboard';
import LobbyLayout from './Lobby';
import LoginLayout from './Login';
import NotFoundPage from './../components/NotFoundPage/NotFoundPage'

import "./AppLayout.css";

const { Header, Content, Footer } = Layout;

class AppLayout extends Component {

    state = {
        collapsed: false,
    };

    onCollapse = (collapsed) => {
        this.setState({ collapsed });
    }
    
    toggle = () => {
        this.setState({
            collapsed: !this.state.collapsed,
        });
    }

    render() {
        return (
            <Router>
                <div>
                    <Layout className="layout">
                        <Header>
                            <div className="logo" />
                            <Menu
                                theme="dark"
                                mode="horizontal"
                                defaultSelectedKeys={['1']}
                                style={{ lineHeight: '64px' }}>
                                <Menu.Item key="1">
                                    <Icon type="cluster" />
                                    <span>Lobby</span>
                                    <Link to="/lobby" />
                                </Menu.Item>
                                <Menu.Item key="2">
                                    <Icon type="desktop" />
                                    <span>Play ToK</span>
                                    <Link to="/game" />
                                </Menu.Item>
                                <Menu.Item key="3">
                                    <Icon type="login" />
                                    <span>Login</span>
                                    <Link to="/login" />
                                </Menu.Item>
                            </Menu>
                        </Header>
                        <Content style={{ margin: '24px 16px', padding: 24, background: '#fff', minHeight: 280 }}>
                            <Switch>
                                <Route exact path="/" component={LobbyLayout} />
                                <Route path="/game" component={GameboardLayout} />
                                <Route path="/lobby" component={LobbyLayout} />
                                <Route path="/login" component={LoginLayout} />
                                <Route component={NotFoundPage} />
                            </Switch>
                        </Content>
                        <Footer style={{ textAlign: 'center' }}>SE691 Software Studio - ToK</Footer>
                    </Layout>
                </div>
            </Router>
        );
    }
}

export default AppLayout;