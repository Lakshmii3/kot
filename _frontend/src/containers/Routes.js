import React, { Component } from 'react';
import {
    BrowserRouter as Router,
    Route,
    Switch
} from 'react-router-dom';

import AppLayout from './AppLayout';
import NotFoundPage from './../components/NotFoundPage/NotFoundPage'

class Routes extends Component {
    render() {
        return (
            <div>
                <Router>
                    <div>
                        <Switch>
                            <Route exact path="/" component={AppLayout} />
                            <Route component={AppLayout} />
                        </Switch>
                    </div>
                </Router>
            </div>
        )
    }
}

export default Routes;