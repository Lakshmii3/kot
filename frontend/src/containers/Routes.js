import React, { Component } from 'react';
import {
    BrowserRouter as Router,
    Route,
    Switch
} from 'react-router-dom';

import AppLayout from './AppLayout';

class Routes extends Component {
    render() {
        return (
            <div>
                <Router>
                    <div>
                        <Switch>
                            <Route exact path="/" component={AppLayout} />
                            <Route component={AppLayout} />
                        </Switch>
                    </div>
                </Router>
            </div>
        )
    }
}

export default Routes;