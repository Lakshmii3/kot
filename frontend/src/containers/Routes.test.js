import React from 'react';
import { shallow, mount } from 'enzyme';

import Routes from './Routes';

import AppLayout from './AppLayout';
import NotFoundPage from './../components/NotFoundPage/NotFoundPage'

import { MemoryRouter } from 'react-router'
import { Route } from 'react-router-dom';

describe('routes using memory router', () => {
  it('should show Home component for / router (using memory router)', () => {
    const component = mount( <MemoryRouter initialEntries = {['/']} >
        <Routes/>
      </MemoryRouter>
    );
    expect(component.find(AppLayout)).toHaveLength(1);
  })
  it('should show No match component for route not defined', () => {
    const component = mount( <MemoryRouter initialEntries = {['/unknown']} >
        <Routes/>
      </MemoryRouter>
    );
    expect(component.find(NotFoundPage)).toHaveLength(1);
  })
})