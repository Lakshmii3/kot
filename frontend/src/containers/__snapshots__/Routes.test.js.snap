// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify routes are reliable Verify root of site render correctly 1`] = `
<div>
  <BrowserRouter>
    <div>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </div>
  </BrowserRouter>
</div>
`;
